src/host/configuration.h

```diff
@@ -74,6 +74,7 @@ namespace host
     ccf::ds::TimeString slow_io_logging_threshold = {"10ms"};
     std::optional<std::string> node_client_interface = std::nullopt;
     ccf::ds::TimeString client_connection_timeout = {"2000ms"};
+    std::optional<ccf::ds::TimeString> idle_connection_timeout = std::nullopt;
```

Comment:
I've wrapped this in `optional` so that we can retain the option for the old behaviour (no automated cleanup), in case someone has a hard-to-fix client with long-lived idle sessions.

Open question is what should the default be? A sensible default value is probably around `60s`, so that most users benefit from this by default, but that's potentially a breaking change?